### PR TITLE
Fixes #9859: Broken directive migration in Rudder 4.0.2

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://www.rudder-project.org/redmine/issues/9859

### CODE REVIEW

File: rudder-web/src/main/scala/com/normation/rudder/web/components/DirectiveEditForm.scala [hunk 1]
```diff
@@ -178,7 +178,15 @@ class DirectiveEditForm(
   val displayDeprecationWarning = technique.deprecrationInfo match {
     case Some(info) =>
       ( "#deprecation-message *" #> info.message &
-        "#migrate-button *" #> migrateButton(fullActiveTechnique.techniques.keys.max,"Migrate now!","deprecation-migration")
+        "#migrate-button *" #> {
+            val lastTechniqueVersion = fullActiveTechnique.techniques.keys.max
```
Comment (fanf) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/DirectiveEditForm.scala]: can't it fail with no such element ?

Comment (VinceMacBuche) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/DirectiveEditForm.scala]: Is it possible that a technique may have no technique version ? 

Here a directive has already been selected meaning that we have already built a directive based on a existing version of the technique.

I guess it's impossible in a normal usage of Rudder, but i think it's still possible by manipulating ldap to have no version and still one in the technique

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
OK, merging this PR

**Pull request merged**